lib/spack/spack/test/conftest.py

```diff
@@ -1706,3 +1706,61 @@ def noncyclical_dir_structure(tmpdir):
         with open(j('file_3'), 'wb'):
             pass
     yield d
+
+
+@pytest.fixture(scope='function')
+def mock_url_config(monkeypatch):
+    """Mock web utility's curl and spider functions related to reading from
+       a URL that might represent a single configuration file or a directory
+       containing multiple configuration files.
+
+       Configuration information will be taken from the corresponding file(s)
+       in the test data configuration directory."""
```

Comment:
I think it would be clearer to rephrase

> Configuration information will be taken from the corresponding file(s) in the test data configuration directory.

as

> This will redirect requested URLs to test/data/config and assumes that they are requesting a config file in that directory (i.e. the basename of the URL matches a file in that directory)

Reply:
Done